File: macos/spotter-macOS/Panel/Panel.m
#import "React/RCTBridgeModule.h"
#import "React/RCTEventEmitter.h"

@interface RCT_EXTERN_MODULE(Panel, RCTEventEmitter)

RCT_EXTERN_METHOD(toggle)

RCT_EXTERN_METHOD(open)

RCT_EXTERN_METHOD(close)

@end

File: macos/spotter-macOS/Panel/Panel.swift
import Foundation
import ShellOut

@objc(Panel)
class Panel: RCTEventEmitter {

  let appDelegate = NSApplication.shared.delegate as! AppDelegate

  override init() {
    super.init()
  }

  @objc func open() {
    DispatchQueue.main.async {
      self.appDelegate.openSpotter()
    }
  }

  @objc func close() {
    DispatchQueue.main.async {
      self.appDelegate.closeSpotter()
    }
  }

  @objc func toggle() {
    DispatchQueue.main.async {
      self.appDelegate.toggleSpotter()
    }
  }
  
  @objc func openSettings() {
    DispatchQueue.main.async {
      self.appDelegate.openSettings()
    }
  }

  override func supportedEvents() -> [String]! {
    return []
  }

  @objc override static func requiresMainQueueSetup() -> Bool {
    return false
  }

}
